rules/Transform/Rector/Class_/AddAllowDynamicPropertiesAttributeRector.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Rector\Transform\Rector\Class_;
+
+use PhpParser\Node;
+use PhpParser\Node\Stmt\Class_;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\ValueObject\PhpVersionFeature;
+use Rector\Php80\NodeAnalyzer\PhpAttributeAnalyzer;
+use Rector\PhpAttribute\Printer\PhpAttributeGroupFactory;
+use Rector\VersionBonding\Contract\MinPhpVersionInterface;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @changelog https://wiki.php.net/rfc/deprecate_dynamic_properties
+ *
+ * @see \Rector\Tests\Transform\Rector\Class_\AddAllowDynamicPropertiesAttributeRector\AddAllowDynamicPropertiesAttributeRectorTest
+ */
+final class AddAllowDynamicPropertiesAttributeRector extends AbstractRector implements MinPhpVersionInterface
+{
+    /**
+     * @var string
+     */
+    private const ATTRIBUTE = 'AllowDynamicProperties';
+
+    public function __construct(
+        private PhpAttributeAnalyzer $phpAttributeAnalyzer,
+        private PhpAttributeGroupFactory $phpAttributeGroupFactory,
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Add the `AllowDynamicProperties` attribute to all classes', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+class SomeObject {
+    public string $someProperty = 'hello world';
+}
+CODE_SAMPLE
+
+                ,
+                <<<'CODE_SAMPLE'
+#[AllowDynamicProperties]
+class SomeObject {
+    public string $someProperty = 'hello world';
+}
+CODE_SAMPLE
+            ),
+        ]);
+    }
+
+
+    public function getNodeTypes(): array
+    {
+        return [Class_::class];
+    }
+
+    /**
+     * @param Class_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (
+            $node->extends !== null && (string) $node->extends === 'stdClass' ||
```

Comment:
you can use : 

```php
use PhpParser\Node\Name\FullyQualified;

if ($node->extends instanceof FullyQualified
 && $this->nodeNameResolver->isName($node->extends, 'stdClass'
```

Reply:
Shouldn't we use reflection here instead? For the classes who inherit stdClass but not directly

Reply:
probably can compare with `FamilyRelationsAnalyzer::getClassLikeAncestorNames`

https://github.com/rectorphp/rector-src/blob/f8e8efa7289090f27ab78dbfac424f07fa5fad8b/packages/FamilyTree/Reflection/FamilyRelationsAnalyzer.php#L126